src/jquery.multiselect.js

```diff
@@ -184,105 +195,125 @@
+      var id = elSelect.id || self.multiselectID;              // unique ID for the label & option tags
+      var inputID = 'ui-multiselect-' + self.multiselectID + '-' + (option.id || id + '-option-' + self.inputIdCounter++);
+      var isMultiple = elSelect.multiple;                      // Pick up the select type from the underlying element
       var isDisabled = option.disabled;
       var isSelected = option.selected;
 
-      var $input = $( document.createElement('input') )
-                           .attr({
-                             "name": "multiselect_" + id,
-                             "type": isMultiple ? 'checkbox' : 'radio',
-                             "value": option.value,
-                             "title": title,
-                             "id": inputID,
-                             "checked": isSelected ? "checked" : null,
-                             "aria-selected": isSelected ? "true" : null,
-                             "disabled": isDisabled ? "disabled" : null,
-                             "aria-disabled": isDisabled ? "true" : null
-                           })
-                           .data(option.dataset);
-
-      var $span = this.options.htmlOptionText 
-                  ? $( document.createElement('span') ).html( option.innerHTML ) 
-                  : $( document.createElement('span') ).text( option.textContent );
-      var optionImageSrc = option.getAttribute('data-image-src');
-      if (optionImageSrc)
-        $span.prepend( $( document.createElement('img') ).attr('src', optionImageSrc) );
-
-     var $label = $( document.createElement('label') )
-         .attr({ 'for': inputID,  'title': title})
-         .addClass( (isDisabled ? 'ui-state-disabled ' : '') 
-                   + (isSelected && !isMultiple ? 'ui-state-active ' : '') 
-                   + 'ui-corner-all')
-         .append($input, $span);
-
-      var $item = $( document.createElement('li') )
-         .addClass( (isDisabled ? 'ui-multiselect-disabled ' : '') + (option.className || '') )
-         .append($label);
-
-      return $item;
+      var input = document.createElement('input');
+      var inputAttribs = {
+                          "type": isMultiple ? 'checkbox' : 'radio',
+                          "id": inputID,
+                          "title": title,
+                          "value": option.value,
+                          "name": self.options.addInputNames ? "multiselect_" + id : null,
+                          "checked": isSelected ? "checked" : null,
+                          "aria-selected": isSelected ? "true" : null,
+                          "disabled": isDisabled ? "disabled" : null,
+                          "aria-disabled": isDisabled ? "true" : null
+                         };
+      for (var name in inputAttribs) {
+         if (inputAttribs[name] !== null)
+            input.setAttribute(name,inputAttribs[name]);
+      }
+      if ('dataset' in option) {
```

Comment:
The dataset attribute isn't supported in IE 10, it only got added in 11 and for the time being I would prefer to maintain IE 10 support.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset

Reply:
Thanks.   Let me see if I can find a work around.